common/models/user.js

```diff
@@ -860,7 +860,7 @@ module.exports = function(User) {
         }
       } else {
         var emailChanged = ctx.hookState.originalUserData.some(function(data) {
-          return data.email != ctx.data.email;
+          return ctx.data.email && data.email != ctx.data.email;
```

Comment:
It seems that with the falsy `ctx.data.email`, you don't even need to iterate over `ctx.hookState.originalUserData`